src/rgw/rgw_common.h

```diff
@@ -1297,6 +1297,7 @@ struct req_state {
   /* aws4 auth support */
   bool aws4_auth_needs_complete;
   bool aws4_auth_streaming_mode;
+  bool aws4_auth_presigned_url_bugfix;
```

Comment:
@jmunhoz it looks like this variable doesn't need to be in req_state itself. consider adding a `bool boto2_compat = false` parameter to `RGW_Auth_S3::authorize_v4()`, then have `RGW_Auth_S3::authorize()` pass in `true` the second time around